packages/block-editor/src/store/private-selectors.js

```diff
@@ -698,3 +698,13 @@ export function getClosestAllowedInsertionPointForPattern(
 	const names = getGrammar( pattern ).map( ( { blockName: name } ) => name );
 	return getClosestAllowedInsertionPoint( state, names, clientId );
 }
+
+/**
+ * Where the point where the next block will be inserted into.
+ *
+ * @param {Object} state
+ * @return {Object} where the insertion point in the block editor is or null if none is set.
+ */
+export function getInsertionPoint( state ) {
```

Comment:
How is this different than `getBlockInsertionPoint`?

Reply:
`getBlockInsertionPoint` is really about the blue insertion cue that gets shown. `getBlockInsertionPoint` is constantly updated based on where you're hovering the canvas. Maybe originally `getBlockInsertionPoint` was about where the insertion would happen, but its code now is tied to where the insertion cue shows.

Reply:
The difference is that one is an index and the other one is an UI element?

Reply:
Both are state for me. The UI element is just a reflexion of a state.

Reply:
I think it gets tricky when you include drag and drop. You’re showing an insertion cue at a location that may be different than what is set via selection. 

If you update that point on hover via drag and drop, this is the correct behavior for the insertion cue/insertion point, but a lot of the things built off using it were not intended to be updating their insertion point that often. 

So, I think insertion cue of where insertion will happen and should rely on insertion point if set, but the insertion cue should be able to show separate from the insertion point on instances like drag and drop. Hovering the canvas should not be updating the insertion point, it should be updating a cue about where you might be able to insert.

Reply:
> So, I think insertion cue of where insertion will happen and should rely on insertion point if set, but the insertion cue should be able to show separate from the insertion point on instances like drag and drop. Hovering the canvas should not be updating the insertion point, it should be updating a cue about where you might be able to insert.

I agree with this, but it doesn't really explain the PR to me. we were using `showInsertionPoint` for me before this PR to do this and this was already different than the "default" insertion point used in the inserter component. 

So I'm still not clear what this PR is trying to accomplish aside moving things around and potentially making them more complicated.